machine_learning/data_downloader.py

```diff
@@ -0,0 +1,147 @@
+    path : string (default : current working directory)
+        Path to the directory where the csv is stored and downloaded content will be stored
+    """
+    if (path == ''):
+        path = os.getcwd()
+    os.mkdir(dirname)
+    new_dir = os.path.join(path, dirname)
+    download_log_handler = open(str(dirname)+ '.txt', 'w')
+    filename = os.path.join(os.getcwd(), filename)
+    os.chdir(new_dir)
+    download_log_handler.write(time.asctime(time.localtime(time.time())) + '\n')
+    df = pd.read_csv(filename, header=0, error_bad_lines=False)
+    urls = df['Last Two - Side by Side']
+
+    for index in range(len(df['Last Two - Side by Side'])):
+        from_version_uri, to_version_uri = get_storage_uris(urls[index])
+
+        if (from_version_uri == None and to_version_uri == None):
+            download_log_handler.write('Change ' + str(index + 1)
+                                       + ' Incorrect urls' + '\n')
+            continue
+
+        from_version_response = requests.get(from_version_uri)
+        to_version_response = requests.get(to_version_uri)
+
+        if (from_version_response.ok and to_version_response.ok):
+            with open('change_' + str(index + 1) + '_from_version.html',
+                      'w', encoding = 'utf-8') as f:
+                f.write(from_version_response.text)
+            with open('change_' + str(index + 1) + '_to_version.html',
+                      'w', encoding = 'utf-8') as f:
+                f.write(to_version_response.text)
+            download_log_handler.write('Change ' + str(index + 1)
+                                       + ' from version uri : '
+                                       + from_version_uri + '\n')
+            download_log_handler.write('Change ' + str(index + 1)
+                                       + ' to version uri : '
+                                       + to_version_uri + '\n')
+        else:
+            if (not from_version_response.ok):
+                download_log_handler.write('Change ' + str(index + 1)
+                                           + ' from version issue :'
+                                           + from_version_response.text)
+            if (not to_version_response.ok):
+                download_log_handler.write('Change ' + str(index + 1)
+                                           + ' to version issue :'
+                                           + to_version_response.text)
+            continue
+
+    download_log_handler.close()
+    os.chdir('..')
+    return True
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Run downloader script')
+    parser.add_argument('filename', type=str, help='CSV file which has the urls')
+    parser.add_argument('dirname', type=str, help='Directory name in which you want to download the content')
+    parser.add_argument('--path', type=str, help='Path where you want to create the new directory', default='')
```

Comment:
@mhucka I think my explanation of the argument confused you. I had written in the comments of the function `"Path to the directory where the csv is stored and downloaded content will be stored"`. 
What I meant to say here was `"Path to the directory where the csv is stored and downloaded content directory will be stored"` or better yet `"Path to the directory where the csv is stored and directory 'dirname' will be created"`. Does that clear the confusion?